File: effecientcal.R
num1 <- readline(prompt = "Enter first number: ")
num2 <- readline(prompt = "Enter second number: ")
operation <- readline(prompt = "Enter operation (add, subtract, multiply, divide): ")
add <- function(x,y){
  num1 <<- x
  num2 <<- y
  return(num1 + num2)
}
subtract <- function(x,y){
  num1 <<- x
  num2 <<- y
  return(num1 - num2)
}
multiply <- function(x,y){
  num1 <<- x
  num2 <<- y
  return(num1 * num2)
}
divide <- function(x,y){
  num1 <<- x
  num2 <<- y
  if(num2 == 0){
    return("Error: Division by zero is not allowed.")
  } else {
    return(num1 / num2)
  }
}

if(operation == "add"){
  result <- add(as.numeric(num1), as.numeric(num2))
  cat("Result:", result)
} else if(operation == "subtract"){
  result <- subtract(as.numeric(num1), as.numeric(num2))
  cat("Result:", result)
} else if(operation == "multiply"){
  result <- multiply(as.numeric(num1), as.numeric(num2))
  cat("Result:", result)
} else if(operation == "divide"){
  result <- divide(as.numeric(num1), as.numeric(num2))
  cat("Result:", result)
} else {
  cat("Invalid operation. Please enter add, subtract, multiply, or divide.")
}

File: effecientcal.pl
#!/usr/bin/perl
$num = 0;
$num1 = 0;
$oprator = "";

print "Enter the first number: ";
$num = <STDIN>;
print "Enter the second number: ";
$num1 = <STDIN>;
print "Enter the oprator: ";
$oprator = <STDIN>;

chomp($num, $num1, $oprator);

if ($oprator eq "+") {
    print "The sum is: " . ($num + $num1);
} elsif ($oprator eq "-") {
    print "The difference is: " . ($num - $num1);
} elsif ($oprator eq "*") {
    print "The product is: " . ($num * $num1);
} elsif ($oprator eq "/") {
    if ($num1 != 0) {
        print "The quotient is: " . ($num / $num1);
    } else {
        print "Error: Division by zero is not allowed.";
    }
} else {
    print "Invalid oprator. Please use +, -, *, or /.";
}